File: DSA/N_PARTH_PANDEY_590012761/Day15Question2.py
def sort_by_length(arr):
    n = len(arr)
    for i in range(n):
        for j in range(0, n - i - 1):
            if len(arr[j]) > len(arr[j + 1]):
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
    return arr


input_strings = ["apple", "pie", "washington", "cat"]
output_strings = sort_by_length(input_strings)
print(output_strings)

File: DSA/Shubham_590016506/Day_15_Question_2.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int compare(const void* a, const void* b) {
    return strlen(*(char**)a) - strlen(*(char**)b);
}

int main() {
    char* arr[] = {"apple", "pie", "washington", "cat"};
    int n = sizeof(arr) / sizeof(arr[0]);
    qsort(arr, n, sizeof(char*), compare);
    
    for (int i = 0; i < n; i++)
        printf("%s ", arr[i]);
    return 0;
}